kayenta-objectstore-configbin/src/main/java/com/netflix/kayenta/configbin/storage/ConfigBinStorageService.java

```diff
@@ -285,27 +285,49 @@ public void deleteObject(String accountName, ObjectType objectType, String objec
       String ownerApp = credentials.getOwnerApp();
       String configType = credentials.getConfigType();
       ConfigBinRemoteService remoteService = credentials.getRemoteService();
-      String jsonBody =
-          AuthenticatedRequest.allowAnonymous(
-              () ->
-                  retry.retry(
-                      () -> remoteService.list(ownerApp, configType), MAX_RETRIES, RETRY_BACKOFF));
-
-      try {
-        List<String> ids =
-            kayentaObjectMapper.readValue(jsonBody, new TypeReference<List<String>>() {});
-
-        if (ids.size() > 0) {
-          return ids.stream().map(i -> metadataFor(credentials, i)).collect(Collectors.toList());
+      List<String> ids = new ArrayList<>();
+      boolean hasNext = true;
+      String pageId = null;
+      while (hasNext) {
```

Comment:
Logic is largely the same here. There is now a response object that wraps the list of ids and the response can be paged.